maestro-orchestra/src/main/java/maestro/orchestra/Orchestra.kt

```diff
@@ -840,21 +840,58 @@ class Orchestra(
                     lookupTimeoutMs
                 }
             )
-
         val (description, filterFunc) = buildFilter(
             selector,
             deviceInfo(),
         )
+        if (selector.childOf != null) {
+            val parentViewHierarchy = findElementViewHierarchy(
+                selector.childOf,
+                timeout
+            )
+            return maestro.findElementWithTimeout(
+                timeout,
+                filterFunc,
+                parentViewHierarchy
+            ) ?: throw MaestroException.ElementNotFound(
+                "Element not found: $description",
+                maestro.viewHierarchy().root,
```

Comment:
```suggestion
                parentViewHierarchy.root
```
Probably should be the parent view hierarchy attached to the error here instead of the root as we can't find the child in the parent hierarchy